Sources/DOT/Graph+Rendering.swift

```diff
@@ -29,10 +29,8 @@ fileprivate func which(_ command: String) throws -> URL {
     return URL(fileURLWithPath: string)
 }
 
-extension Graph {
-    public func render(using layout: LayoutAlgorithm, to format: Format) throws -> Data {
-        let encoded = DOTEncoder().encode(self)
-
+extension String {
+    public func renderAsDOT(using layout: LayoutAlgorithm, to format: Format) throws -> Data {
```

Comment:
`String` is a commonly used type in Swift, and adding a GraphViz-specific convenience method in an extension has the effect of polluting the API surface area with unrelated functionality.

If the intention is to decouple graph rendering to allow for raw DOT to be rendered by `dot`, then I think a better design would be to create a `Renderer` type that can process both `Graph` and `String` values. The advantage there is that a `Renderer` type could be responsible for formats and layout algorithms, which is currently delegated to `Graph` in a way that arguably conflates model and view concerns.

Reply:
Fixed!